pkg/provisioner/provisionerdependencies.go

```diff
@@ -0,0 +1,81 @@
+package provisioner
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+)
+
+const (
+	dependencyRetryDelay = time.Second * 10
+)
+
+// DependencyStatus describes the current availability status of a generic dependency
+// required by the provisioner to operate correctly
+type DependencyStatus struct {
+	// The name of the dependency
+	Name string
+	// If ready, the dependency is available to process incoming requests
+	Ready bool
+	// Can provide additional info if the dependency is not ready
+	Err error
+}
+
+// Dependency represents a required entity by the Provisioner
+type Dependency interface {
+	// The name of the dependency
+	Name() string
+	// This method verifies the current status of the dependency
+	Check(done chan<- DependencyStatus)
+}
+
+// DependenciesChecker manages a collection of Provisioner dependencies
+type DependenciesChecker struct {
+	deps []Dependency
+	log  logr.Logger
+}
+
+// CheckDependencies verifies in parallel all the registered dependencies and
+// waits for the results. If one of the dependencies has an error then a requeue result
+// is returned
+func (p *DependenciesChecker) CheckDependencies() (result Result, err error) {
+
+	p.log.Info("verifying provisioner dependencies")
+
+	var done = make(chan DependencyStatus, len(p.deps))
+	defer close(done)
+
+	for _, dep := range p.deps {
+		go dep.Check(done)
```

Comment:
I think this should just be a single, simple for loop and remove all the concurrency. 

I don't see what happens if one of your dep checks gets stuck -- there's no context allowing you to cancel the goroutine so you could get stuck in your loop below reading from the channels. 

See: https://rakyll.org/leakingctx/, https://dave.cheney.net/2016/12/22/never-start-a-goroutine-without-knowing-how-it-will-stop

Reply:
In this case all the checks were simple http requests with timeout. In any case as per previous comments I'm going to refactor it in a simpler manner and remove all the goroutines